File: oldcontest/edits/solutions/carlo.cpp
//Author: Carlo Malagnino

//Task: edits

#pragma GCC optimize("Ofast")
#include<bits/stdc++.h>

using namespace std;

int main(){
    int T;
    string ln;
    scanf("%d", &T);

    for(int t=1;t<=T;t++){
        cin >> ln;
        int p=0, c=0;
        string cmd;
        int n=ln.length();
        while(p<n){
            if (ln[p]=='['){
                if(ln[p+1]=='C'){
                    if(c>0){
                        cmd=cmd.substr(0,c-1)+cmd.substr(c);
                        c-=1;
                    }
                    p+=3;
                    continue;
                }
                if(ln[p+1]=='L'){
                    c=max(0,c-1);
                    p+=3;
                    continue;
                }
                c=min(int(cmd.length()),c+1);
                p+=3;
                continue;
            }
            cmd=cmd.substr(0,c)+ln[p]+cmd.substr(c);
            p+=1;
            c+=1;
        }
        cout << "Case #"<< t << ": " << cmd << "\n";
    }
}

File: contest21old/edits/solutions/carlo.py
"""
@author: Carlo Malagnino

Task: Edits
"""

from blist import blist

T=int(input())

for t in range(1,T+1):
    ln=input()
    ln=input()
    n=len(ln)
    p=0
    cmd=blist()
    c=0
    while(p<n):
        if ln[p]=="[":
            if(ln[p+1]=="C"):
                if(c>0):
                    del(cmd[c-1])
                    c-=1
            elif(ln[p+1]=="L"):
                c=max(0,c-1)
            else:
                c=min(len(cmd),c+1)
            p+=3
        else:
            cmd.insert(c,ln[p])
            p+=1
            c+=1
    print("Case #{}: {}".format(t,''.join(cmd)))
